### PR TITLE
better docs for ActiveRecord::AttributeSet::YAMLEncoder#decode

### PR DESCRIPTION
**Pull request by taboularasa**

### CODE REVIEW

File: activerecord/lib/active_record/attribute_set/yaml_encoder.rb [hunk 1]
```diff
@@ -7,6 +7,10 @@ def initialize(default_types)
         @default_types = default_types
       end
 
+      # Populate +coder+ with 'concise_attributes' using +attribute_set+.
+      #
+      # If any attribute's name matches a +default_types+ key then the attribute
```
Comment (maclover7) [inline activerecord/lib/active_record/attribute_set/yaml_encoder.rb]: comma after key

File: activerecord/lib/active_record/attribute_set/yaml_encoder.rb [hunk 2]
```diff
@@ -7,6 +7,10 @@ def initialize(default_types)
...
+      # will be cast by passing +nil+ to ActiveRecord::Attribute#with_type
```
Comment (maclover7) [inline activerecord/lib/active_record/attribute_set/yaml_encoder.rb]: period after `with_type`.

File: activerecord/lib/active_record/attribute_set/yaml_encoder.rb [hunk 3]
```diff
@@ -17,6 +21,18 @@ def encode(attribute_set, coder)
         end
       end
 
+      # Decode a previously encoded +coder+.
+      #
+      # +coder+ should be the result of previously encoding an Active Record
+      # model, using #encode.
+      #
+      # If +coder+ contains an 'attributes' key then #decode returns the value
```
Comment (maclover7) [inline activerecord/lib/active_record/attribute_set/yaml_encoder.rb]: comma after key

File: activerecord/lib/active_record/attribute_set/yaml_encoder.rb [hunk 4]
```diff
@@ -17,6 +21,18 @@ def encode(attribute_set, coder)
...
+      # under 'attributes' key.
+      #
+      # Otherwise decode will build an attributes hash from coder using the
```
Comment (maclover7) [inline activerecord/lib/active_record/attribute_set/yaml_encoder.rb]: comma after decode

File: activerecord/lib/active_record/attribute_set/yaml_encoder.rb [hunk 5]
```diff
@@ -17,6 +21,18 @@ def encode(attribute_set, coder)
...
+      # value under 'concise_attributes' while casting any nil types using
+      # default_types. The resulting hash is used to return a new
+      # ActiveRecord::AttributeSet
```
Comment (maclover7) [inline activerecord/lib/active_record/attribute_set/yaml_encoder.rb]: period at end of sentence

**Comment by taboularasa** [comment]
Thanks for the feedback, @maclover7 
All comments have been addressed.

**Comment by maclover7** [comment]
Can you squash your commits and add `[ci skip]`?

**Comment by taboularasa** [comment]
@maclover7 oh right, done. Thanks again.

File: activerecord/lib/active_record/attribute_set/yaml_encoder.rb [hunk 6]
```diff
@@ -17,6 +22,18 @@ def encode(attribute_set, coder)
         end
       end
 
+      # Decode a previously encoded +coder+.
+      #
+      # +coder+ should be the result of previously encoding an Active Record
+      # model, using #encode.
+      #
+      # If +coder+ contains an 'attributes' key, then #decode returns the value
+      # under 'attributes' key.
```
Comment (maclover7) [inline activerecord/lib/active_record/attribute_set/yaml_encoder.rb]: under the

File: activerecord/lib/active_record/attribute_set/yaml_encoder.rb [hunk 7]
```diff
@@ -17,6 +22,18 @@ def encode(attribute_set, coder)
...
+      #
+      # Otherwise, decode will build an attributes hash from coder using the
```
Comment (maclover7) [inline activerecord/lib/active_record/attribute_set/yaml_encoder.rb]: #decode

File: activerecord/lib/active_record/attribute_set/yaml_encoder.rb [hunk 8]
```diff
@@ -17,6 +22,18 @@ def encode(attribute_set, coder)
...
+      # value under 'concise_attributes' while casting any nil types using
```
Comment (maclover7) [inline activerecord/lib/active_record/attribute_set/yaml_encoder.rb]: comma after the last quotation mark, before the while

**Comment by maclover7** [comment]
Sorry, found a few more things to change 😬 

Thanks again for submitting.

**Comment by taboularasa** [comment]
@maclover7 no problem at all, thanks for all the feedback, I'm just getting started with documenting Rails but I intend to spend a lot of time doing this so I'm happy to get all the feedback I can get. I'll work on this at lunch

**Comment by taboularasa** [comment]
@maclover7 updated again.

**Comment by maclover7** [comment]
:+1: Thank you for your contribution!

**Comment by maclover7** [comment]
PR merged @ be8b2ddbcb66babd768ee756604e60e71060ecd9.

:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:

**Comment by taboularasa** [comment]
👌

**Pull request closed without merging**